### PR TITLE
Publish a ClusterOperator with conditions for status

### PR DESCRIPTION
**Pull request by dgoodwin**
This will expand once we have error conditions on CredentialsRequests.

### CODE REVIEW

File: pkg/controller/credentialsrequest/status.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	cloudCredOperatorNamespace = "openshift-cloud-credential-operator"
+)
+
+// syncOperatorStatus computes the operator's current status and therefrom
```
Comment (joelddiaz) [inline pkg/controller/credentialsrequest/status.go]: s/therefrom/from there/ ?

File: pkg/controller/credentialsrequest/status.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+	for _, cr := range credRequests {
+		if !cr.Status.Provisioned {
+			credRequestsNotProvisioned = credRequestsNotProvisioned + 1
+		}
+	}
+	if credRequestsNotProvisioned > 0 {
+		progressingCondition.Status = configv1.ConditionTrue
+		progressingCondition.Reason = "Reconciling"
+		progressingCondition.Message = fmt.Sprintf(
+			"%d/%d credentials requests provisioned",
```
Comment (joelddiaz) [inline pkg/controller/credentialsrequest/status.go]: maybe "%d of %d credentials..." instead of the '/' to be more clear

**Comment by dgoodwin** [comment]
Updated, thanks!

**Comment by dgoodwin** [comment]
ready for another look @joelddiaz

**Comment by dgoodwin** [comment]
Should be ready for another round of review.

File: pkg/controller/credentialsrequest/status.go [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+	}
+	failingCredRequests := 0
+
+	for _, cr := range credRequests {
+		// Check for provision failure conditions:
+		foundFailure := false
+		for _, t := range failureConditionTypes {
+			failureCond := utils.FindCredentialsRequestCondition(cr.Status.Conditions, t)
+			if failureCond != nil && failureCond.Status == corev1.ConditionTrue {
+				foundFailure = true
```
Comment (csrwng) [inline pkg/controller/credentialsrequest/status.go]: really small nit: break here

Comment (dgoodwin) [reply pkg/controller/credentialsrequest/status.go]: done!

**Comment by dgoodwin** [comment]
Finally passed, @csrwng can I get anotherr lgtm, quickly. :) There was a change to add a missing list permission for caching.

File: Dockerfile [hunk 4]
```diff
@@ -10,6 +10,7 @@ WORKDIR /go/src/github.com/openshift/cloud-credential-operator
 COPY pkg/    pkg/
```
Comment (abhinavdahiya) [inline Dockerfile]: why not `COPY . .` ?

File: pkg/util/clusteroperator/status.go [hunk 5]
```diff
@@ -0,0 +1,65 @@
+package clusteroperator
```
Comment (abhinavdahiya) [inline pkg/util/clusteroperator/status.go]: you can use `https://github.com/openshift/library-go/tree/master/pkg/config/clusteroperator/v1helpers`

**Comment by dgoodwin** [comment]
@abhinavdahiya will look into those as a followup, this has been open too long and need to get it merged if CI will work. (spoiler alert, it won't)

**Pull request merged**